src/tests/test_config.cpp

```diff
@@ -286,4 +289,58 @@ BOOST_AUTO_TEST_CASE ( test_variable_info )
 	}
 }
 
+/**
+ * Test if config::add_child(key) correctly adds a child with the key.
+ * @pre The constructors of config are correct.
+ */
+BOOST_AUTO_TEST_CASE ( test_add_child__key )
```

Comment:
In C++, identifies with a double-underscore are reserved, even when it's in the middle of the identifier.

Reply:
I fixed the name and pass the test parameters directly into the test code.